api/redhatopenshift/HcpCluster.Management/hcpOperatorIdentityRoles.tsp

```diff
@@ -0,0 +1,82 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+namespace Microsoft.RedHatOpenShift;
+
+using Azure.ResourceManager;
+
+/** HcpOperatorIdentityRoles represents a location based representation of
+ * the required platform workload identities and their required roles for a given
+ * OpenShift version
+ */
+@locationResource
+model HcpOperatorIdentityRoleSet
+  is ProxyResource<HcpOperatorIdentityRoleSetProperties> {
+  ...ResourceNameParameter<
+    HcpOperatorIdentityRoleSet,
+    NamePattern = "^\\d+\\.\\d+"
```

Comment:
Since this is a read-only resource do we _need_ to restrict the name pattern?

For installable OCP versions ([XCMSTRAT-613](https://issues.redhat.com/browse/XCMSTRAT-613)) was thinking of having a special resource named "default" in addition to the "X.Y.Z" resource names.  I don't know if we want to have something similar for this resource type, but maybe we shouldn't preclude it in the API spec?